File: test/java/nz/co/searchwellington/spam/SpamFilterTest.java
package nz.co.searchwellington.spam;

import nz.co.searchwellington.model.Resource;
import nz.co.searchwellington.model.WebsiteImpl;
import junit.framework.TestCase;

public class SpamFilterTest extends TestCase {
    
    SpamFilter filter = new SpamFilter();
    
    public void testAllowsNormalSubmission() throws Exception {        
        Resource okResource = new WebsiteImpl();
        okResource.setName("Test site");
        okResource.setUrl("http://www.test.com.localhost");
        okResource.setDescription("test test");        
        assertFalse(filter.isSpam(okResource));               
    }
    
    public void testShouldBlockRFID() throws Exception {        
        Resource spamResource = new WebsiteImpl();
        spamResource.setName("Test site");
        spamResource.setUrl("http://www.rfid.com");
        spamResource.setDescription("test test");      
        assertTrue(filter.isSpam(spamResource));               
    }
    
    public void testShouldBlockByDescription() throws Exception {        
        Resource spamResource = new WebsiteImpl();
        spamResource.setName("Test site");
        spamResource.setUrl("http://www.test.com.localhost");
        spamResource.setDescription("test rfid test");      
        assertTrue(filter.isSpam(spamResource));               
    }

}

File: src/test/scala/nz/co/searchwellington/spam/SpamFilterTest.scala
package nz.co.searchwellington.spam

import java.util.UUID

import nz.co.searchwellington.model.Website
import org.junit.jupiter.api.Assertions._
import org.junit.jupiter.api.Test

class SpamFilterTest {

  private val filter = new SpamFilter

  @Test
  def shouldAllowNormalSubmission(): Unit = {
    val okResource = Website(id = UUID.randomUUID().toString, title = "Test site", page = "http://www.test.com.localhost", description = Some("test test"))

    assertFalse(filter.isSpam(okResource))
  }

  @Test
  def testShouldBlockRFIDUrl(): Unit = {
    val spamResource = Website(id = UUID.randomUUID().toString, title = "Test site", page = "http://www.rfid.com", description = Some("test test"))

    assertTrue(filter.isSpam(spamResource))
  }

  @Test
  def testShouldBlockByDescription(): Unit = {
    val spamResource = Website(id = UUID.randomUUID().toString, title = "Test site", page = "http://www.test.com.localhost", description = Some("test rfid test"))

    assertTrue(filter.isSpam(spamResource))
  }

}